### PR TITLE
feat(sakavottord2): new application type

### PR DESCRIPTION
**Pull request by johannaagma**
# ...

Attach a link to issue if relevant

## What

A new application in the application system where people can apply for their criminal record. 

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by darrikonn — COMMENTED** [review]
Can you remove the `kubectl` binary?

File: package.json
Comment (darrikonn) [inline package.json:142]: as far as I can see, this is not used

Comment (darrikonn) [reply package.json:142]: maybe a peer dep of react-pdf?

File: apps/service-portal/src/assets/images/bus.svg
Comment (thordurhhh) [inline apps/service-portal/src/assets/images/bus.svg:1]: This file seems to be still in use in the service-portal. 😱

Comment (johannaagma) [reply apps/service-portal/src/assets/images/bus.svg]: Seems like the file changed from 433 lines to 1 line (but we didnt change anything), but I have reverted the changes in this file

File: package.json
Comment (johannaagma) [reply package.json]: We have removed this dependency

**Review by darrikonn — COMMENTED** [review]
Can you add your team to the CODEOWNERS file as well.

**Comment by johannaagma** [comment]
@darrikonn We have already been added to codeowners here:
https://github.com/island-is/island.is/pull/5873/files

File: libs/api/domains/criminal-record/src/lib/graphql/main.resolver.ts
Comment (darrikonn) [inline libs/api/domains/criminal-record/src/lib/graphql/main.resolver.ts:20]: Both of these queries look dangerous.
By implementing it like this you're exposing every single criminal record to users that have a valid login. It's most always incorrect to pass in the nationalId as an argument.
In addition, this is not following the naming conventions for GraphQL, ref: https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions

Arguments should be input objects and names `input`. The name of the queries should be the domain itself:
`CriminalRecord`. And this should most likely use the CurrentUser as the nationalId in order to prevent exposing the pre-mentioned danger ^.

**Review by darrikonn — CHANGES_REQUESTED** [review]

File: libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx
Comment (stjanilofts) [inline libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx:3]: I don't know for sure but according to [this](https://github.com/wojtekmaj/react-pdf#webpack) I think you can skip `pdfjs` for the worker stuff by importing this way:

```javascript
import { Document, Page } from 'react-pdf/dist/esm/entry.webpack';
```

**Review by stjanilofts — APPROVED** [review]
Approved (for island-ui codeowners) but with one question.

File: libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx
Comment (unnursolingimars) [reply libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx:3]: Yes we actually imported it from 'react-pdf/dist/esm/entry.webpack' first but got some errors on the automatic checks so we changed it in consultation with Eiríkur :)

**Review by Ingimarsson — APPROVED** [review]
Changes in syslumenn client look good.

File: libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts
Comment (baering) [inline libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts:47]: Intentionally no title here?

File: libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts
Comment (baering) [inline libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts:90]: Missing translations here.

File: libs/application/templates/criminal-record/src/lib/CriminalRecordTemplate.ts
Comment (baering) [inline libs/application/templates/criminal-record/src/lib/CriminalRecordTemplate.ts:137]: You only want to return this role if the `nationalId` (passed in as the first argument) is equal to `application.applicant` (application is passed in as the second argument).

File: libs/application/templates/criminal-record/src/forms/Payment.tsx
Comment (baering) [inline libs/application/templates/criminal-record/src/forms/Payment.tsx:39]: I think this will break the application and likely require manual database state change so that the user can interact with it again.

Comment (baering) [reply libs/application/templates/criminal-record/src/forms/Payment.tsx:39]: You might want to conditionally show a screen in case the url is missing to allow the user to move the application to draft and try again. Similarly only showing this screen when the url does exist.

**Review by baering — COMMENTED** [review]
Good job on your first application. I did see some things that need to be addressed and then left some minor comments as well.

File: libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts
Comment (darrikonn) [inline libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts:6]: Wondering about this approach.
This GraphQL field is not used so it doesn't really matter (might be worth removing it). But do we plan on being able to download a Criminal Record file, like a PDF?

Another possible solution instead of returning base64 encoded file (which has a huge payload), is to use the download service. Might be worth checking out if you plan on being able to download the file via browser?

File: libs/clients/criminal-record/src/lib/criminalRecordApi.service.ts
Comment (darrikonn) [inline libs/clients/criminal-record/src/lib/criminalRecordApi.service.ts:14]: Another solution would be to upload it to the download S3 bucket... but might be better to look at the download service?

**Review by darrikonn — COMMENTED** [review]
Looking good!
Core files mostly approved, just waiting for answers regarding my PR review questions. Otherwise approved 👍

File: libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts
Comment (unnursolingimars) [reply libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts:47]: Yes it's supposed to be hidden from the user because it contains no personal data.

File: libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts
Comment (unnursolingimars) [reply libs/application/templates/criminal-record/src/forms/CriminalRecordForm.ts:90]: Thanks! Fixing it.

File: libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts
Comment (johannaagma) [reply libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts]: Ah forgot to remove this model class, after I removed an unecessary function (getCriminalRecord) in the graphQL layer.

But we are going to allow users to download the criminal record as PDF (through libs/client, not libs/api - if that matters, so it is not going through the graphQL layer).
I hadn't realized we had a download service, but we also need to display the PDF in the browser, so it will look something like this:
image

Is it possible to use this download service to access the base64 string for the PDF to display in the browser (as well as being able to download the file)?

File: libs/application/templates/criminal-record/src/forms/Payment.tsx
Comment (unnursolingimars) [reply libs/application/templates/criminal-record/src/forms/Payment.tsx:39]: Okay thanks, fixing it :)

File: libs/application/templates/criminal-record/src/dataProviders/FeeInfoProvider.ts
Comment (unnursolingimars) [reply libs/application/templates/criminal-record/src/dataProviders/FeeInfoProvider.ts:9]: @baering Do you have a suggestion of where I should store these values?

File: libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts
Comment (darrikonn) [reply libs/api/domains/criminal-record/src/lib/graphql/models/criminalRecord.model.ts]: Alright, nice!

Yeah I believe you can use it.. it's just a REST API. Implementation can be seen in `apps/download-service/*`, although you probably won't need it since if your interaction is through a REST API you can just stream it yourself :).

**Review by darrikonn — APPROVED** [review]
LGTM

**Comment by johannaagma** [comment]
@Ingimarsson We made some more small changes to Syslumenn client, you can see them here: https://github.com/island-is/island.is/pull/5834/commits/012b0288a12edfe53ba1f5c083025cbe99cd9bf6
Could you review that as well? :)

File: libs/application/templates/criminal-record/src/dataProviders/FeeInfoProvider.ts
Comment (baering) [reply libs/application/templates/criminal-record/src/dataProviders/FeeInfoProvider.ts:9]: Maybe add this as an enum to a new constant file here `libs/shared/constants/src/lib`

**Review by baering — APPROVED** [review]
LGTM

**Pull request merged**
